source/index.js

```diff
@@ -147,7 +174,19 @@ module.exports = async (input = 'patch', options) => {
 						return `Private package: not publishing to ${pkgManagerName}.`;
 					}
 				},
-				task: (context, task) => publishTaskHelper(pkgManager, task, options, input)
+				task: (context, task) => {
+					try {
+						return publishTaskHelper(pkgManager, task, options, input)
+							.pipe(
+								catchError(rollback)
+							)
+							.subscribe(() => {
```

Comment:
You can't subscribe. What you are returning to the task is the subscription which isn't an observable anymore.

I think what you want is something like this

```js
task: (context, task) => {
	let hasError = false;

	return publishTaskHelper(pkgManager, task, options, input)
		.pipe(
			catchError(() => {
				hasError = true;

				return rollback();
			}),
			finalize(() => {
				isPublished = !hasError;
			})
		)
}
```

Reply:
Yes, that looks good 👍🏻 Thanks for the help 🙌🏻